src/huggingface_hub/hf_api.py

```diff
@@ -764,9 +776,19 @@ def list_datasets(
                 The limit on the number of datasets fetched. Leaving this option to `None` fetches all datasets.
             full (:obj:`bool`, `optional`):
                 Whether to fetch all dataset data, including the `lastModified` and the `cardData`.
-
+            use_auth_token (:obj:`bool` or :obj:`str`, `optional`):
+                Whether to use the `auth_token` provided from the `huggingface_hub` cli. If not logged in,
+                a valid `auth_token` can be passed in as a string.
         """
         path = f"{self.endpoint}/api/datasets"
+        if use_auth_token:
+            if isinstance(use_auth_token, str):
+                if not self._is_valid_token(use_auth_token):
+                    raise ValueError("Invalid token passed!")
+                token = use_auth_token
+            else:
+                token = HfFolder.get_token()
```

Comment:
Consider something like this for consistency

```suggestion
                token = HfFolder.get_token()
                if token is None:
                  raise EnvironmentError(
                    "You need to provide a `token` or be logged in to Hugging Face with "
                    "`huggingface-cli login`."
                  )
```

Reply:
Opted for this error as it's a bit more specific towards what's happening:
```python
                        "You need to provide a `token` to `use_auth_token` or be logged in to Hugging Face with "
                        "`huggingface-cli login`."
```

Reply:
You're right